### PR TITLE
Timeout for download and request

### LINKED ISSUE #129: Timeout for getting files from IPFS
Hi,

Is there any way to specify timeout for methods like cat()/get() or any other way to handle calls to this functions with hash to non-existing resource?

Thanks

### PR DESCRIPTION
**Pull request by AliabbasMerchant**
Fixes #129

### CODE REVIEW

**Comment by alexander255** [comment]
Looks good! Two questions however:

  * Is 120 the previous default of `requests` or was that another, different, value?
  * Could you check out the build failures on CI? I didn't say anything yet, because I thought that you would fix them.

**Comment by AliabbasMerchant** [comment]
The default value of python `request` is None. I added the 120 value by myself. IPFS sometimes takes some time to connect, so I thought 120 should be okay. Do you want me to change it to something else?

**Comment by AliabbasMerchant** [comment]
@alexander255 I think there is a prob with the multipart.py file.
The CI was failing even in PR #141 , in which I hadn't even changed any code file.    
Also, there is a problem with Windows compatability.  
I'll look into the CI failures once you are done with PR #145

**Comment by hareeshnagaraj** [comment]
@AliabbasMerchant - Thanks for making this fix and opening the PR! We actually have the exact same timeout requirement, it is good to know we are not the only ones

**Comment by AliabbasMerchant** [comment]
@alexander255   
Can you please merge this??   
I have written the corres. tests also, for the timeout requirements. They are working properly.  
The CI errors are occurring due to some other reason (I suspect `multipart.py`)

**Comment by alexander255** [comment]
Sorry about this! I've put merging stuff on hold until your renaming changes were in, so to not cause you extra conflicts while changing everything. Could you please reopen this PR against the `py-ipfs-http-client` branch? I know the current situation is a bit of a mess, but it'll get better from now on. Promise!

**Comment by AliabbasMerchant** [comment]
@alexander255 
I changed the base branch of this PR to `py-ipfs-http-client`

**Comment by AliabbasMerchant** [comment]
No. The timeout tests were passing that time.

File: test/unit/test_http.py [hunk 1]
```diff
@@ -45,15 +42,32 @@ def return_okay(url, request):
...
+    url -- the url of the incoming request
+    request -- the request that is being responded to
+    """
+    return {
+        'status_code': 500,
+        'content': 'fail'.encode('utf-8'),
+    }
+
 
+@urlmatch(netloc='localhost:5001', path=r'.*/fail')
```
Comment (alexander255) [inline test/unit/test_http.py]: The problem is that this matches `/fail`, but your test expects `/timeout`. Please change this to the later and the tests will likely pass. The codestyle issue will disappear if you rebase onto current the current `py-ipfs-http-client` commit, so please do that as well as there are merge-conflicts atm.

**Comment by alexander255** [comment]
@AliabbasMerchant: Will you push an update for this and rebase it onto latest `py-ipfs-http-client` so that we can ship it? :slightly_smiling_face:

**Comment by parkan** [comment]
@alexander255 deprecated in favor of #169, let's close this one :)

**Comment by alexander255** [comment]
@parkan: Thanks for noticing!

**Pull request closed without merging**